6.0.0 changelog and new file (#8020)
hashicorp/terraform-provider-google-beta: CHANGELOG.md

## 6.0.0 (Unreleased)

[Terraform Google Provider 6.0.0 Upgrade Guide](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_6_upgrade)

BREAKING CHANGES:
* provider: changed provider labels to add the `goog-terraform-provisioned: true` label by default. ([#8004](https://github.com/hashicorp/terraform-provider-google-beta/pull/8004))

> * activedirectory: added `deletion_protection` field to `google_active_directory_domain` to make deleting them require an explicit intent. `google_active_directory_domain` resources now cannot be destroyed unless `deletion_protection = false` is set for the resource. ([#7837](https://github.com/hashicorp/terraform-provider-google-beta/pull/7837))

Reviewer: ```suggestion
* * activedirectory: added `deletion_protection` field to `google_active_directory_domain` resource. This field defaults to `true`, preventing accidental deletions. To delete a `google_active_directory_domain` resource, you must first set `deletion_protection = false` and apply this change before destroying the resource. ([#7837](https://github.com/hashicorp/terraform-provider-google-beta/pull/7837))
```

* alloydb: removed `network` in `google_alloy_db_cluster`. Use `network_config.network` instead. ([#7999](https://github.com/hashicorp/terraform-provider-google-beta/pull/7999))

> * billing: updated resource id for `google_billing_project_info` ([#7793](https://github.com/hashicorp/terraform-provider-google-beta/pull/7793)) 

Reviewer: ```suggestion
* billing: revised the format of `id` for `google_billing_project_info` ([#7793](https://github.com/hashicorp/terraform-provider-google-beta/pull/7793)) 
```

* bigquery: added validation to prevent table view creation if schema contains required fields for `google_bigquery_table` resource ([#7755](https://github.com/hashicorp/terraform-provider-google-beta/pull/7755))
* bigquery: removed `allow_resource_tags_on_deletion` from `google_bigquery_table`. Resource tags are now always allowed on table deletion. ([#7940](https://github.com/hashicorp/terraform-provider-google-beta/pull/7940))
* bigqueryreservation: removed `multi_region_auxiliary` from `google_bigquery_reservation` ([#7844](https://github.com/hashicorp/terraform-provider-google-beta/pull/7844))

> * cloudrunv2: added `deletion_protection` field to `google_cloudrunv2_service` to make deleting them require an explicit intent. `google_cloudrunv2_service` resources now cannot be destroyed unless `deletion_protection = false` is set for the resource. ([#7901](https://github.com/hashicorp/terraform-provider-google-beta/pull/7901))

Reviewer: recommend consistency with deletion protection notes.

* identityplatform: removed resource `google_identity_platform_project_default_config` in favor of `google_identity_platform_project_config` ([#7880](https://github.com/hashicorp/terraform-provider-google-beta/pull/7880))
* integrations: removed `create_sample_workflows` and `provision_gmek` from `google_integrations_client` ([#7977](https://github.com/hashicorp/terraform-provider-google-beta/pull/7977))
* pubsub: allowed `schema_settings` in `google_pubsub_topic` to be removed ([#7674](https://github.com/hashicorp/terraform-provider-google-beta/pull/7674))
* redis: added a `deletion_protection_enabled` field with a default value of `true` to the `google_redis_cluster` resource ([#7995](https://github.com/hashicorp/terraform-provider-google-beta/pull/7995))
* resourcemanager: added `deletion_protection` field to `google_folder` to make deleting them require an explicit intent. Folder resources now cannot be destroyed unless `deletion_protection = false` is set for the resource. ([#7903](https://github.com/hashicorp/terraform-provider-google-beta/pull/7903))

> * resourcemanager: made `deletion_policy` in `google_project` 'PREVENT' by default. This makes deleting them require an explicit intent. `google_project` resources cannot be destroyed unless `deletion_policy` is set to 'ABANDON' or 'NONE' for the resource. ([#7946](https://github.com/hashicorp/terraform-provider-google-beta/pull/7946))

Reviewer 2: ```suggestion
* resourcemanager: made `deletion_policy` in `google_project` 'PREVENT' by default. This makes deleting them require an explicit intent. `google_project` resources cannot be destroyed unless `deletion_policy` is set to 'ABANDON' or 'DELETE' for the resource. ([#7946](https://github.com/hashicorp/terraform-provider-google-beta/pull/7946))
```

Reviewer 2: Sorry, I forgot to modify the release note in the related PR.

* storage: removed `no_age` field from  `lifecycle_rule.condition` in the `google_storage_bucket` resource ([#7923](https://github.com/hashicorp/terraform-provider-google-beta/pull/7923))
* sql: removed `settings.ip_configuration.require_ssl` within `google_sql_database_instance`. Please use `settings.ip_configuration.ssl_mode` instead. ([#7804](https://github.com/hashicorp/terraform-provider-google-beta/pull/7804))
* vpcaccess: removed default values for `min_throughput` and `min_instances` fields on `google_vpc_access_connector` and made them default to values returned from the API when not provided by users ([#7709](https://github.com/hashicorp/terraform-provider-google-beta/pull/7709))
* vpcaccess: added a conflicting fields restriction between `min_throughput` and `min_instances` fields on `google_vpc_access_connector` ([#7709](https://github.com/hashicorp/terraform-provider-google-beta/pull/7709))

> * vpcaccess: added a conflicting fields restriction between `max_throughput` and `max_instances` fields on `google_vpc_access_connector`

Author: ```suggestion
* vpcaccess: added a conflicting fields restriction between `max_throughput` and `max_instances` fields on `google_vpc_access_connector` ([#7709](https://github.com/hashicorp/terraform-provider-google-beta/pull/7709))
```

> * workstation: defaulted `host.gce_instance.disable_ssh` to true for `google_workstations_workstation_config` ([#7946](https://github.com/hashicorp/terraform-provider-google-beta/pull/7709 https://github.com/hashicorp/terraform-provider-google-beta/pull/7946))

Author: ```suggestion
* workstation: defaulted `host.gce_instance.disable_ssh` to true for `google_workstations_workstation_config` ([#7946](https://github.com/hashicorp/terraform-provider-google-beta/pull/7946))
```

IMPROVEMENTS:

> * compute: added fields `reserved_internal_range` and `secondary_ip_ranges[].reserved_internal_range` to `google_compute_subnetwork` resource

Author: ```suggestion
* compute: added fields `reserved_internal_range` and `secondary_ip_ranges[].reserved_internal_range` to `google_compute_subnetwork` resource ([7980](https://github.com/hashicorp/terraform-provider-google-beta/pull/7980))
```

> * compute: changed the behavior of `name_prefix` in multiple Compute resources to allow for a longer max length of 54 characters. See the upgrade guide and resource documentation for more details. ([#7980](https://github.com/hashicorp/terraform-provider-google-beta/pull/7980 https://github.com/hashicorp/terraform-provider-google-beta/pull/7980))

Author: ```suggestion
* compute: changed the behavior of `name_prefix` in multiple Compute resources to allow for a longer max length of 54 characters. See the upgrade guide and resource documentation for more details. ([#7981](https://github.com/hashicorp/terraform-provider-google-beta/pull/7981))
```
